packages/mobile/src/components/track-list/TrackList.tsx

```diff
@@ -70,16 +91,42 @@ export const TrackList = ({
           />
         ) : null}
         <TrackListItem
-          {...track}
-          key={track.trackId}
           index={index}
+          drag={drag}
+          hideArt={hideArt}
+          isActive={isActive}
+          isDragging={isDragActive}
+          isPlaying={isPlaying}
+          isReorderable={isReorderable}
+          track={track}
+          key={track.track_id}
           onSave={onSave}
           togglePlay={togglePlay}
           trackItemAction={trackItemAction}
+          onRemove={onRemove}
         />
       </View>
     )
   }
 
-  return <FlatList data={tracks} renderItem={renderTrack} />
+  return (
+    <DraggableFlatList
+      activationDistance={scrollEnable ? 100 : 1}
+      data={tracks.filter(filterFn ?? (() => true))}
+      keyExtractor={(track, index) => `${track.track_id} ${index}`}
+      onDragBegin={() => {
```

Comment:
would we always want haptics for onDragBegin?

Reply:
Yes, at least that's how the current one works. Because it's a long press to start the drag, it's kinda important to have that feedback when the dragging state begins